### PR TITLE
Feature/actionbutton

### PR DESCRIPTION
**Pull request by AllTwos**
## What?
I've added a small button/icon that performs unique actions while keeping count. 

## Why?
This button/icon can be used for simple things such as: likes, comments, etc.

## How?
The icon itself is passed as a prop with an id='kid', therefore enabling usage in styled.css

## Testing?
N/A

## Screenshots (optional)
0

## Anything Else?
N/A

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -14,6 +17,21 @@ function App() {
         onSubmit={() => console.log('submitted')}
       />
       <LongButton onClick={() => null}>Example</LongButton>
+      <ActionButton
```
Comment (souzasmatheus) [inline src/App.tsx]: From now on, let's remove these components from the App.tsx file once we finish building them. That is, we surely need to import them here for now to see how they render, but let's not commit them here. We have to treat the develop branch like our production code.

File: src/App.tsx [hunk 2]
```diff
@@ -14,6 +17,21 @@ function App() {
...
+        onPress={() => console.log('clicked ActionButton')}
+        icon={<FaHeart id="kid" />}
```
Comment (souzasmatheus) [inline src/App.tsx]: This was a wise solution, but let's reflect about it for a while:

We now have to pass a `id="kid"` every time we wish to use this component. There are a few problems with this approach. It is repetitive and not really documented (a new person would have to go to our codebase and see this component being used in order to know they should use an id, they would have to remember it and it isn't even that clear why we need it, unless they check out the component's styles).

A better solution could be to create a component which automatically creates an icon with the `id="kid"` to the icon. But, again, how to dynamically pass this icon prop? I will give you a suggestion bellow.

File: src/components/common/atoms/ActionButton/index.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  counter?: number | string;
+  icon: ReactNode;
+  onPress: () => void;
+};
+
+const ActionButton = ({ counter, icon, onPress }: Props) => {
+  return (
+    <Wrapper>
+      <StyledIcon onClick={onPress}>{icon}</StyledIcon>
+      <StyledCount>{!counter ? 0 : counter}</StyledCount>
```
Comment (souzasmatheus) [inline src/components/common/atoms/ActionButton/index.tsx]: `counter ?? 0`

Checkout this out: [Nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)

**Review by souzasmatheus — CHANGES_REQUESTED** [review]

File: src/components/common/atoms/ActionButton/index.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { ReactNode } from 'react';
+import { StyledIcon, Wrapper, StyledCount } from './styled';
+
+type Props = {
+  counter?: number | string;
+  icon: ReactNode;
+  onPress: () => void;
```
Comment (souzasmatheus) [inline src/components/common/atoms/ActionButton/index.tsx]: Let's keep the convention an call it `onClick`

File: src/App.tsx (hunk 1 above)
Comment (AllTwos) [reply src/App.tsx]: I AM THE BEST!

File: src/components/common/atoms/ActionButton/index.tsx (hunk 3 above)
Comment (AllTwos) [reply src/components/common/atoms/ActionButton/index.tsx]: did this I WIN I ALWAYS WIN

File: src/components/common/atoms/ActionButton/index.tsx (hunk 4 above)
Comment (AllTwos) [reply src/components/common/atoms/ActionButton/index.tsx]: CHOKE WASN'T GOOD ENOUGH

**Review by souzasmatheus — APPROVED** [review]

**Pull request merged**